synapse/python_dependencies.py

```diff
@@ -89,7 +89,7 @@
         # python 3.5.2, as per https://github.com/itamarst/eliot/issues/418
         'eliot<1.8.0;python_version<"3.5.3"',
     ],
-    "saml2": ["pysaml2>=4.5.0"],
+    "saml2": ["pysaml2>=4.9.0"],
```

Comment:
no particular reason but 4.5.0 has been released in october 2017, and according to release notes, 4.9.0 don't change any API. So I assume it would have been better to bump the minor version for stability improvement. Up to you, I can revert

Reply:
generally we don't bump our minimum required dependencies unless there is a need to solve a particular problem. Bumping the version for no reason makes things difficult for distribution package maintainers.